site/docs/hosting/vercel.md

```diff
@@ -0,0 +1,106 @@
+# Hosting: Vercel Serverless Functions
+
+This tutorial will guide you on how to deploy your bot to [Vercel](https://vercel.com/) by using [Vercel Serverless Functions](https://vercel.com/docs/concepts/functions/serverless-functions), assuming that you already have a [Vercel](https://vercel.com) account.
+
+## Project Structure
+
+The only prerequisite to get started with **Vercel Serverless Functions** is to move your code to the `api/` directory as shown below. You can also see [Vercel's documentation](https://vercel.com/docs/concepts/functions/serverless-functions#deploying-serverless-functions) for more on this.
+
+```asciiart:no-line-numbers
+.
+├── node_modules/
+├── build/
+├── api/
+│   └── bot.ts
+├── package.json
+├── package-lock.json
+└── tsconfig.json
+```
+
+If you are using TypeScript, you might as well want to install `@vercel/node` as a dev dependency, but it is not mandatory to follow this guide.
+
+## Configuring Vercel
+
+The next step is to create a `vercel.json` file at the top level of your project. For our example structure, its content would look be:
+
+```json
+{
+  "functions": {
+    "api/bot.ts": {
+      "memory": 1024,
+      "maxDuration": 10
+    }
+  }
+}
+```
+
+> Vercel's free subscription has restrictions on quotas, which we need to enable in the configuration file, `vercel.json` ([grammY Examples](https://github.com/grammyjs/examples/blob/main/vercel-bot/api/index.ts)).
+
+If you want to learn more about this configuration file, see [its documentation](https://vercel.com/docs/project-configuration).
+
+## Configuring TypeScript
+
+In our `tsconfig.json` we have to define specify our output directory as `build/`\*, and our root directory is `api/`. This is important since we will specify them in Vercel's deploy options.
```

Comment:
```suggestion
In our `tsconfig.json` we have to define specify our output directory as `build/`, and our root directory is `api/`. This is important since we will specify them in Vercel's deploy options.
```